x-pack/plugins/osquery/server/usage/fetchers.ts

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  SingleBucketAggregate,
+  TopHitsAggregate,
+  ValueAggregate,
+} from '@elastic/elasticsearch/api/types';
+import { PackagePolicyServiceInterface } from '../../../fleet/server';
+import { getRouteMetric } from '../routes/usage';
+import { ElasticsearchClient, SavedObjectsClientContract } from '../../../../../src/core/server';
+import { ListResult, PackagePolicy, PACKAGE_POLICY_SAVED_OBJECT_TYPE } from '../../../fleet/common';
+import { OSQUERY_INTEGRATION_NAME } from '../../common';
+import { METRICS_INDICES } from './constants';
+import { AgentInfo, BeatMetricsUsage, LiveQueryUsage } from './types';
+
+interface PolicyLevelUsage {
+  scheduled_queries?: ScheduledQueryUsageMetrics;
+  agent_info?: AgentInfo;
+}
+
+export async function getPolicyLevelUsage(
+  esClient: ElasticsearchClient,
+  soClient: SavedObjectsClientContract,
+  packagePolicyService?: PackagePolicyServiceInterface
+): Promise<PolicyLevelUsage> {
+  if (!packagePolicyService) {
+    return {};
+  }
+  const packagePolicies = await packagePolicyService.list(soClient, {
+    kuery: `${PACKAGE_POLICY_SAVED_OBJECT_TYPE}.package.name:${OSQUERY_INTEGRATION_NAME}`,
+    perPage: 10_000,
+  });
+
+  const result: PolicyLevelUsage = {
+    scheduled_queries: getScheduledQueryUsage(packagePolicies),
+    // TODO: figure out how to support dynamic keys in metrics
+    // packageVersions: getPackageVersions(packagePolicies),
+  };
+  const agentResponse = await esClient.search({
+    body: {
+      size: 0,
+      aggs: {
+        policied: {
+          filter: {
+            terms: {
+              policy_id: packagePolicies.items.map((p) => p.policy_id),
+            },
+          },
+        },
+      },
+    },
+    index: '.fleet-agents',
+  });
+  const policied = agentResponse.body.aggregations?.policied as SingleBucketAggregate;
+  if (policied && typeof policied.doc_count === 'number') {
+    result.agent_info = {
+      enrolled: policied.doc_count,
```

Comment:
This has given me some good ideas on something I'm working on at the moment. I think this is good.